docs/pages/07_service.adoc

```diff
@@ -0,0 +1,1092 @@
+
+
+==== @New
+
+Use `@New` annotation on a function to define a function that creates a new transfer object. The function creates the new transfer object as defined by its return type. The new transfer object will not be persisted, only the transfer object is returned.
+
+*Example:*
+
+----
+@New
+function InvoiceTransfer myNew();
+----
+
+@New annotation can be used only on functions that
+
+* have return type
+* have no parameter
+* have no body
+
+
+==== @Create and @Mapped
+
+Use `@Create` annotation on a function to define a create function. The create function persists the mapped transfer object passed as its argument. The transfer object's mapped entity must not exist. It persists the transfer object by creating a new entity.
+
+The `@Create` annotation persists only the fields of the transfer object that have the `@Mapped` annotation.
+
+*Example:*
+
+----
+entity Invoice {
+    field String customerName;
+    field String comment;
+}
+
+transfer InvoiceTransfer(Invoice invoice) {
+    @Mapped
+    field String customerName = invoice.customerName;
+    field String comment = invoice.comment;
+}
+
+service InvoiceService {
+    @Create
+    function myCreate(InvoiceTransfer invoiceTransfer);
+}
+----
+
+The above example declares the myCreate function, which creates a new Invoice entity and sets only the entity's customerName field. The comment field is not persisted because the comment field of the transfer object is not marked with `@Mapped`.
+
+@Create annotation can be used only on functions that
+
+* have a mapped transfer object parameter
+* have no return type
+* have no body
+
+
+==== @Update and @Mapped
+
+Use `@Update` annotation on a function to define a new update function. The update function persists the mapped transfer object passed as its argument.  The transfer object's mapped entity must already exist.
+
+The `@Update` annotation persists only the fields of the transfer object that have the `@Mapped` annotation.
+
+*Example:*
+
+
+----
+entity Invoice {
+    field String customerName;
+    field String comment;
+}
+
+transfer InvoiceTransfer(Invoice invoice) {
+    @Mapped
+    field String customerName = invoice.customerName;
+    field String comment = invoice.comment;
+}
+
+service InvoiceService {
+    @Update
+    function myUpdate(InvoiceTransfer invoiceTransfer);
+}
+----
+
+The above example declares the myUpdate function, which sets only the entity's customerName field. The comment field is not persisted because the comment field of the transfer object is not marked with `@Mapped`.
+
+@Update annotation can be used only on functions that
+
+* have a mapped transfer object parameter
+* have no return type
+* have no body
+
+
+==== @Delete
+
+Use `@Delete` annotation on a function to define a new delete function. The delete function deletes the entity referenced by its export's mapping field.
```

Comment:
Did we mean "by its service mapping field"?

Reply:
ok